src/gui/generalsettings.cpp

```diff
@@ -370,24 +374,31 @@ void GeneralSettings::slotUpdateChannelChanged()
         case 2:
             return QStringLiteral("daily");
             break;
+        case 3:
+            return QStringLiteral("enterprise");
+            break;
         default:
             return QStringLiteral("stable");
         }
     };
 
+    ConfigFile configFile;
     const auto channel = updateChannelFromLocalized(_ui->updateChannel->currentIndex());
-    if (channel == ConfigFile().currentUpdateChannel()) {
+    if (channel == configFile.currentUpdateChannel()) {
         return;
     }
 
+    const auto enterprise = configFile.validUpdateChannels().contains("enterprise") ? "- enterprise: contains stable versions for customers.\n"
+                                                                                    : "";
     auto msgBox = new QMessageBox(
         QMessageBox::Warning,
         tr("Changing update channel?"),
         tr("The channel determines which upgrades will be offered to install:\n"
            "- stable: contains tested versions considered reliable\n"
            "- beta: contains versions with new features that may not be tested thoroughly\n"
            "- daily: contains versions created daily only for testing and development\n\n"
-           "Downgrading versions is not possible immediately: changing from beta to stable means waiting for the new stable version."),
+           "%1\n"
+           "Downgrading versions is not possible immediately: changing from beta to stable means waiting for the new stable version.").arg(enterprise),
```

Comment:
from a translation point of view, that will not work and the *enterprise* release channel description will go untranslated
remove the `\n` at the end of this line `"- daily: contains versions created daily only for testing and development\n\n"`, you already have one after the `%1` and with empty placeholder, that would be 3 line breaks when before that was 2
maybe you should add a `tr` when creating `enterprise` string and add additional context for translators about the %1 placeholder